src/pages/billing/List/CreateBillingProjectControl.ts

```diff
@@ -0,0 +1,41 @@
+import { Fragment } from 'react'
+import { h, span } from 'react-hyperscript-helpers'
+import { ButtonOutline } from 'src/components/common'
+import { icon } from 'src/components/icons'
+import { MenuButton } from 'src/components/MenuButton'
+import { MenuTrigger } from 'src/components/PopupTrigger'
+import { CloudProvider } from 'src/libs/workspace-utils'
+
+
+interface CreateBillingProjectControlProps {
+  isAzurePreviewUser: boolean
+  showCreateProjectModal: (type: CloudProvider) => void
+}
+
+export const CreateBillingProjectControl = (props: CreateBillingProjectControlProps) => {
+  const createButton = (type: CloudProvider | undefined) => {
+    return h(ButtonOutline, {
+      'aria-label': 'Create new billing project',
+      onClick: () => type !== undefined ? props.showCreateProjectModal(type) : undefined
+    }, [span([icon('plus-circle', { style: { marginRight: '1ch' } }), 'Create'])])
+  }
+
+  if (!props.isAzurePreviewUser) {
+    return createButton('GCP')
```

Comment:
Is there a meaningful difference between `'GCP'` and `cloudProviderTypes.GCP`? To my untrained eye it looks like we're switching away from an enum in favor of a string, but I'm not positive that I'm reading it correctly.

Reply:
TypeScript enforces that the strings are only the declared allowable types. I believe that our preference is to use the strings for brevity. @nawatts is this correct?

![image](https://user-images.githubusercontent.com/484484/225364553-f1e0e7f6-ebee-49fe-802e-8957fe7b4895.png)

Reply:
We decided to use literal unions instead of enums for defining these sort of "enums".
```ts
type CloudProvider = 'AZURE' | 'GCP'
```
vs
```ts
enum CloudProvider {
  AZURE = 'AZURE'
  GCP = 'GCP'
}
```

I don't know if we discussed a convention for passing arguments of enum types. In my opinion, using a string 'GCP' here instead of `cloudProviderTypes.GCP` is fine because (as you point out) TS checks that the argument is valid.

In the pre-TS days, those enum-like objects such as `cloudProviderTypes` served as effectively a type hint for developers, indicating that the function accepted one of those values vs an arbitrary string.